app/views/helpers/sentences.php

```diff
@@ -133,9 +138,50 @@ public function displayTranslations($id, $translations, $indirectTranslations, $
                     $langFilter
                 );
             }
-
-            // indirect translations
-            foreach ($indirectTranslations as $translation) {
+            $visible = 0;
+            if(count($keys)<5){
+                $listed = count($keys);
+           // indirect translations
+                $keys=array_keys($indirectTranslations);
+                for ($i=0; $i < 5-$listed && $i < count($keys); $i++) {
+                    $key=$keys[$i];
+                    $translation = $indirectTranslations[$key];
+                    $this->displayGenericSentence(
+                        $translation['Translation'],
+                        null,
+                        'indirectTranslation',
+                        $withAudio,
+                        $id,
+                        null,
+                        false,
+                        $langFilter
+                    );
+                    $visible++;
+                }
+            }
+            echo '<a id="translation-show-'.$id.'" class="showLink" onclick="toggle_visibility('.$id.');">&#x2193; More Translations</a></div>';
+            //expanded list of translations
+            echo '<div id="translation-'.$id.'" class="more">';
+            $keys=array_keys($translations);
+            for ($i=5; $i < count($keys); $i++) {
```

Comment:
put 5 as a constant so that you dont have to repeat this magic number everywhere